library/core/src/time.rs

```diff
@@ -308,6 +308,29 @@ impl Duration {
         Duration { secs, nanos: subsec_nanos }
     }
 
+
+    /// creates a new Duration from the specified number of nano seconds. 
+    /// Use this function if you need to specify time greater than what can fit in u64 
+    /// which is around 584 years and above 584.94 years to be a little more precise. 
+    /// 
+    /// # Examples 
+    /// 
+    /// ```
+    /// use std::time::Duration;
+    /// let time_in_nanos = 2.pow(64);
+    /// let duration = Duration::from_nanos_u128(time_in_nanos);
+    /// 
+    /// ```
+    pub const fn from_nanos_u128(nanos: u128) -> Duration {
+        const NANOS_PER_SEC: u128 = self::NANOS_PER_SEC as u128;
+        let secs = nanos / NANOS_PER_SEC;
```

Comment:
Miri seems to have a [solution](https://github.com/rust-lang/miri/blob/cffbc13f5ce138dcbc13dcbbe682419ffa796a05/src/clock.rs#L49-L56) for handling cases when the number of seconds is greater than `u64::MAX`.

Maybe a better way to do this is (might need confirmation from someone who would know better):
```suggestion
        let secs = u64::try_from(duration / self::NANOS_PER_SEC).unwrap_or(u64::MAX);
```

Reply:
The ACP and Tracking issue contain a doc commend specifying panic on overflow instead of saturating on overflow.
Also its a bit weird that on overflow the seconds saturate but the nanoseconds aren't saturated as well.